src/EditorFeatures/Core/BackgroundWorkIndicator/BackgroundWorkIndicatorContext.cs

```diff
@@ -89,212 +51,116 @@ public BackgroundWorkIndicatorContext(
-            Contract.ThrowIfTrue(requests.Count > 2, "At most we can have two requests in the queue (one to update, one to dismiss).");
-            Contract.ThrowIfFalse(
-                requests.Contains(UIUpdateRequest.DismissTooltip) || requests.Contains(UIUpdateRequest.UpdateTooltip),
-                "We didn't get an actual event we know about.");
-
-            return requests.Contains(UIUpdateRequest.DismissTooltip)
-                ? DismissUIAsync()
-                : UpdateUIAsync();
-
-            async ValueTask DismissUIAsync()
-            {
-                await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-
-                // Ensure we only dismiss once.
-                if (_dismissed)
-                    return;
-
-                _dismissed = true;
-
-                // Unhook any event handlers we've setup.
-                //
-                // note we have to disconnect from dismissal notifications so our own dismiss below doesn't cause us to
-                // re-enter and cancel ourselves.
-                _toolTipPresenter.Dismissed -= OnToolTipPresenterDismissed;
-                _subjectBuffer.Changed -= OnTextBufferChanged;
-                _textView.LostAggregateFocus -= OnTextViewLostAggregateFocus;
-
-                // Finally, dismiss the actual tool-tip.
-                _toolTipPresenter.Dismiss();
-
-                // Let our factory know that we were disposed so it can let go of us as well.
-                _factory.OnContextDisposed(this);
-            }
-
-            async ValueTask UpdateUIAsync()
-            {
-                // Build the current description in the background, then switch to the UI thread to actually update the
-                // tool-tip with it.
-                var data = this.BuildData();
-
-                await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-
-                // If we've been dismissed already, then no point in continuing.
-                if (_dismissed)
-                    return;
-
-                // Todo: build a richer tool-tip that makes use of things like the progress reported, and perhaps has a
-                // close button.
-                _toolTipPresenter.StartOrUpdate(
-                    _trackingSpan, [string.Format(EditorFeaturesResources._0_Esc_to_cancel, data.description)]);
-            }
+            _backgroundWorkIndicator.Dispose();
+            _factory.OnContextDisposed(this);
         }
 
+        /// <summary>
+        /// The same as Dispose.  Anyone taking ownership of this context wants to show their own UI, so we can just
+        /// close ours.
+        /// </summary>
+        public void TakeOwnership()
+            => this.Dispose();
+
         public IUIThreadOperationScope AddScope(bool allowCancellation, string description)
```

Comment:
>AddScope [](http://example.com/codeflow?start=39&length=8)

Might be worth a comment why AddScope doesn't need to call ReportTotalProgress as it's not obvious to me.

Reply:
sure.  basically,  the reason is simply that when a scope is created, it has no progress (it's effective 0 completed of 0 total).  So it would have no impact on any progress display.  happy to comment that.